syntax = "proto3";
package abilityUnit;
option java_package = "ability.proto";
option java_outer_classname = "AbilityProto";
option java_multiple_files = true;

service Ability{
  rpc Start(StartInfo) returns (Response){}
  rpc Connect (ConnectInfo) returns (Response) {}
  rpc Disconnect(DisconnectInfo) returns (Response){}
  rpc Terminate(TerminateInfo) returns (Response){}
}

message Response{
  int32 code = 1;
  string msg = 2;
}

message StartInfo{
    int32 timestamp = 1;
}

message ConnectInfo{
    string ip = 1;
    int32 port = 2;
    int32 timestamp = 3;
}

message DisconnectInfo{
    int32 timestamp = 1;
}

message TerminateInfo{
    int32 timestamp = 1;
}

